### PR TITLE
Fix `<RadioButtonGroupInput options>` prop was ignored

### PR DESCRIPTION
**Pull request by fzaninotto**
- [x] Fix `<RadioButtonGroupInput options>` prop was ignored
- [x] Rewrite the docs for choice inputs (`<SelectInput>`, `<AutocompleteInput>`, and `<RadioButtonGroupInput>`)
- [x] Modernize the doc for `<ReferenceInput>`
- [x] Add a chapter for `useInput`
- [x] Add a component zoo for inputs

image

### CODE REVIEW

File: docs/RadioButtonGroupInput.md [hunk 1]
```diff
@@ -5,27 +5,52 @@ title: "The RadioButtonGroupInput Component"
...
+ - [`<AutocompleteInput>`](./AutocompleteInput.md) renders a list of suggestions in an autocomplete input
+
+**Tip**: If you need to let users select more than one item in the list, check out the [`<CheckboxGroupInput>`](./CheckboxGroupInput.md) component.
+
+## Props
 
 | Prop              | Required | Type                       | Default | Description                                                                                                                            |
 | ----------------- | -------- | -------------------------- | ------- | -------------------------------------------------------------------------------------------------------------------------------------- |
-| `choices`         | Required | `Object[]`                 | -       | List of items to show as options                                                                                                       |
+| `choices`         | Optional | `Object[]`                 | -       | List of items to show as options. Required unless inside a ReferenceInput.s                                                                                                       |
```
Comment (slax57) [inline docs/RadioButtonGroupInput.md:53]:
```suggestion
| `choices`         | Optional | `Object[]`                 | -       | List of items to show as options. Required unless inside a ReferenceInput.                                                                                                        |
```

File: docs/RadioButtonGroupInput.md [hunk 2]
```diff
@@ -34,68 +59,96 @@ import { RadioButtonGroupInput } from 'react-admin';
...
-   { id: 'M', name: 'myroot.gender.male' },
-   { id: 'F', name: 'myroot.gender.female' },
-];
+<ReferenceInput label="Author" source="author_id" reference="authors">
+    <RadioButtonGroupInput />
+</ReferenceInput>
 ```
 
-However, in some cases (e.g. inside a `<ReferenceInput>`), you may not want the choice to be translated. In that case, set the `translateChoice` prop to `false`.
+See [Using in a `ReferenceInput>`](#using-in-a-referenceinput) below for more information.
```
Comment (slax57) [inline docs/RadioButtonGroupInput.md:111]:
```suggestion
See [Using in a `<ReferenceInput>`](#using-in-a-referenceinput) below for more information.
```

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.stories.tsx [hunk 3]
```diff
@@ -10,14 +10,42 @@ import { FormInspector } from './common.stories';
...
+
+export const DefaultValue = () => (
+    <Wrapper>
+        <RadioButtonGroupInput
+            choices={[
+                { id: 'M', name: 'Male' },
+                { id: 'F', name: 'Female' },
+            ]}
+            source="gender"
+        />
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/RadioButtonGroupInput.stories.tsx:48]: Did you forget to set the default value? Or do I miss the point of this Story?

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.tsx [hunk 4]
```diff
@@ -95,6 +95,7 @@ export const RadioButtonGroupInput = (props: RadioButtonGroupInputProps) => {
         margin = 'dense',
         onBlur,
         onChange,
+        options,
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/RadioButtonGroupInput.tsx:98]: This is a code fix and not a Doc fix -> PR name should reflect that

**Review by slax57 — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.tsx (hunk 4 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/input/RadioButtonGroupInput.tsx:98]: got it, I changed the PR title

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.stories.tsx (hunk 3 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/input/RadioButtonGroupInput.stories.tsx:48]: No, it's just that I replaced the previous "Basic" story with the same code as in the doc, to be able to make screenshots. I kept the previous Basic story and renamed it to DefaultValue, but it's useless and brings confusion, so I'll remove it.

Comment (fzaninotto) [reply packages/ra-ui-materialui/src/input/RadioButtonGroupInput.stories.tsx:48]: Or better, add a DefaultValue

**Review by slax57 — APPROVED** [review]

File: docs/RadioButtonGroupInput.md [hunk 5]
```diff
@@ -34,68 +59,96 @@ import { RadioButtonGroupInput } from 'react-admin';
...
+const ucfirst = name => name.charAt(0).toUpperCase() + name.slice(1);
+const choices = possibleValues.map(value => ({ id: value, name: ucfirst(value) }));
+
+<RadioButtonGroupInput source="category" choices={choices} />
 ```
 
-Lastly, use the `options` attribute if you want to override any of MUI's `<RadioButtonGroup>` attributes:
+## `options`
+
+Use the `options` attribute if you want to override any of MUI's `<RadioButtonGroup>` attributes:
```
Comment (WiXSL) [inline docs/RadioButtonGroupInput.md:125]:
```suggestion
Use the `options` attribute if you want to override any of MUI's `<RadioGroup>` attributes:
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/RadioButtonGroupInput.md [hunk 6]
```diff
@@ -5,27 +5,52 @@ title: "The RadioButtonGroupInput Component"
...
+    category: 'lifestyle',
+}
+```
+
+**Tip**: React-admin includes other components to edit such values:
+
+ - [`<SelectInput>`](./SelectInput.md) renders a dropdown
+ - [`<AutocompleteInput>`](./AutocompleteInput.md) renders a list of suggestions in an autocomplete input
+
+**Tip**: If you need to let users select more than one item in the list, check out the [`<CheckboxGroupInput>`](./CheckboxGroupInput.md) component.
```
Comment (WiXSL) [inline docs/RadioButtonGroupInput.md:47]: What about mentioning the `AutocompleteArrayInput` here as well?

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/Inputs.md [hunk 7]
```diff
@@ -54,7 +54,35 @@ All input components accept the following props:
...
+| Array of objects      | `[{ item: 'jeans', qty: 3 }, { item: 'shirt', qty: 1 }]`     | [`<ArrayInput>`](./ArrayInput.md)                                                                                                                                                                    |
+| Array of Enums        | `['foo', 'bar']`                                             | [`<SelectArrayInput>`](./SelectArrayInput.md), [`<AutocompleteArrayInput>`](./AutocompleteArrayInput.md), [`<CheckboxGroupInput>`](./CheckboxGroupInput.md), [`<DualListInput>`](./DualListInput.md) |
+| Array of foreign keys | `[42, 43]`                                                   | [`<ReferenceArrayInput>`](./ReferenceArrayInput.md)                                                                                                                                                  |
+| Translations          | `{ en: 'Hello', fr: 'Bonjour' }`                             | [`<TranslatedInputs>`](./TranslatedInputs.md)                                                                                                                                                        |
```
Comment (WiXSL) [inline docs/Inputs.md:82]:
```suggestion
| Translations          | `{ en: 'Hello', fr: 'Bonjour' }`                             | [`<TranslatableInputs>`](./TranslatableInputs.md)                                                                                                                                                        |
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/useInput.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+            label={props.label}
+            error={(isTouched || isSubmitted) && invalid}
+            helperText={(isTouched || isSubmitted) && invalid ? error : ''}
+            required={isRequired}
+            {...rest}
+        />
+    );
+};
+const LatLngInput = props => {
+    const {source, ...rest} = props;
```
Comment (WiXSL) [inline docs/useInput.md:91]:
```suggestion
    const { source, ...rest } = props;
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/ReferenceInput.md [hunk 9]
```diff
@@ -5,40 +5,105 @@ title: "The ReferenceInput Component"
...
+            <TextInput source="last_name" />
+            <ReferenceInput source="company_id" reference="companies" />
+        </SimpleForm>
+    </Edit>
+);
+```
+
+`<ReferenceInput>` require a `source` and a `reference` prop.
+
+`<ReferenceInput>` uses the foreign key value to fetch the the related record. It also grabs the list of possible choices for the field. For instance, if the `ContactEdit` component above is used to edit the following contact:
```
Comment (WiXSL) [inline docs/ReferenceInput.md:45]:
```suggestion
`<ReferenceInput>` uses the foreign key value to fetch the related record. It also grabs the list of possible choices for the field. For instance, if the `ContactEdit` component above is used to edit the following contact:
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/ReferenceInput.md [hunk 10]
```diff
@@ -173,29 +240,63 @@ import { ReferenceInput, SelectInput } from 'react-admin';
...
+
+```jsx
+<ReferenceInput source="company_id" reference="companies">
+    <AutocompleteInput optionText="reference" />
+</ReferenceInput>
+```
+
+## `queryOptions`
+
+Use the `queryOptions` prop pass options to the `dataProvider.getList()` query that fetches the possible choices.
```
Comment (WiXSL) [inline docs/ReferenceInput.md:274]:
```suggestion
Use the `queryOptions` prop to pass options to the `dataProvider.getList()` query that fetches the possible choices.
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/SelectInput.md [hunk 11]
```diff
@@ -50,31 +71,40 @@ An array of objects that represents the choices to show in the dropdown. The obj
...
     source="category"
-    optionValue="name"
-    optionText="label"
     choices={choices}
+    optionText="label"
+    optionValue="_id"
 />
 ```
 
+You can render some of the options as disabled by setting the `disabled` field in some of the choices:
```
Comment (WiXSL) [inline docs/SelectInput.md:97]:
```suggestion
You can render some options as disabled by setting the `disabled` field in some choices:
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/AutocompleteInput.md [hunk 12]
```diff
@@ -6,54 +6,69 @@ title: "The AutocompleteInput Component"
...
+It renders using [MUI's `<Autocomplete>`](https://mui.com/components/autocomplete/).
 
 ![AutocompleteInput](./img/autocomplete-input.gif)
 
-Set the `choices` attribute to determine the options list. The value must be an array of objects exposing the `id` and `name` for each choice.
+This input allows editing record fields that are scalar values, e.g. `123`, `'admin'`, etc. 
+
+## Usage
+
+In addition to the `source`, `<AutocompleteInput>` requires one prop: the `choices` listing the possible values.
```
Comment (WiXSL) [inline docs/AutocompleteInput.md:17]: Actually, `choices` is optional, and we say that in the props table below

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/AutocompleteInput.md [hunk 13]
```diff
@@ -98,36 +124,118 @@ When used inside a `<ReferenceInput>`, `<AutocompleteInput>` doesn't need a `cho
...
+        <SimpleForm>
+            <TextInput source="title" />
+            <ReferenceInput source="category_id" reference="categories">
+                <AutocompleteInput create={<CreateCategory />} />
+            </ReferenceInput>
+        </SimpleForm>
+    </Create>
+);
+
+// in ./CreateCategory.js
```
Comment (WiXSL) [inline docs/AutocompleteInput.md:156]:
```suggestion
// in ./CreateCategory.js
import React from 'react';
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

**Review by WiXSL — APPROVED** [review]

File: docs/AutocompleteInput.md (hunk 12 above)
Comment (fzaninotto) [reply docs/AutocompleteInput.md:17]: In the general case, `choices` is required. I detail the general case in the Usage section, then the specific cases in each prop description. For simplicity's sake, I prefer to leave it like that.

**Pull request merged**